src/kirin/dialects/scf/scf2cf.py

```diff
@@ -175,11 +175,7 @@ def rewrite_ssacfg(self, node: ir.Region):
         return RewriteResult(has_done_something=has_done_something)
 
     def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
-        if (
-            isinstance(node, (For, IfElse))
-            or not node.has_trait(ir.HasCFG)
-            and not node.has_trait(ir.SSACFG)
```

Comment:
can you elaborate how does this fix the issue?

Reply:
This is unrelated, it's issue #444. The scf2cf is not compatible with v0.17 if we leave those in, in that the traits `HasCFG` and `SSACFG` just don't exist in v0.17, so I removed those.

Sorry for mixing those commits. You could consider rebasing the branch without squashing so we have them separately.

Reply:
I see that's ok, this is 0.17 specific. I was just wondering.